clang/lib/CodeGen/CGExprScalar.cpp

```diff
@@ -454,6 +471,11 @@ class ScalarExprEmitter
   Value *VisitFixedPointLiteral(const FixedPointLiteral *E) {
     return Builder.getInt(E->getValue());
   }
+
+  Value *VisitDecimalFloatLiteral(const DecimalFloatLiteral *E) {
+    return Builder.getInt(E->getValue());
```

Comment:
If I'm reading the code right, this will return a `llvm::ConstantInt` and thereby lose track of the fact that the value corresponds to a DFP representation. Do we not need to keep track of that somehow so that the constant value gets associated with a DFP IR type? That seems important if we're going to utilize the `fadd` IR instruction/operator.